back-end/src/main/java/kr/co/ssalon/domain/service/QrService.java

```diff
@@ -0,0 +1,122 @@
+package kr.co.ssalon.domain.service;
+
+import com.google.zxing.BarcodeFormat;
+import com.google.zxing.MultiFormatWriter;
+import com.google.zxing.client.j2se.MatrixToImageWriter;
+import com.google.zxing.common.BitMatrix;
+import kr.co.ssalon.domain.entity.Meeting;
+import kr.co.ssalon.domain.entity.Member;
+import kr.co.ssalon.domain.entity.MemberMeeting;
+import kr.co.ssalon.domain.entity.QrLink;
+import kr.co.ssalon.domain.repository.MeetingRepository;
+import kr.co.ssalon.domain.repository.MemberMeetingRepository;
+import kr.co.ssalon.domain.repository.MemberRepository;
+import kr.co.ssalon.oauth2.CustomOAuth2Member;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.multipart.MultipartFile;
+import org.apache.coyote.BadRequestException;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+
+@Service
+@RestController
+@Transactional
+public class QrService {
+
+    @Autowired
+    private RedisTemplate<String, byte[]> redisTemplate;
+
+    @Autowired
+    private MeetingRepository meetingRepository;
+
+    @Autowired
+    private MemberRepository memberRepository;
+
+    @Autowired
+    private MemberMeetingRepository memberMeetingRepository;
+
+    @Autowired
+    private MemberService memberService;
+
+    @Autowired
+    private MeetingService meetingService;
```

Comment:
Autowired를 통한 의존성 주입은 여러가지 측면에서 지양해야 하는 것으로 알고 있습니다. Autowired 주입과 생성자 주입의 차이점을 한 번 보시는 것도 좋겠네요. 저희 프로젝트의 다른 Service를 보시면 S3 부분을 제외하고 생성자 주입을 사용하고 있습니다. 혹시 이 QR 관련된 부분은 Autowired를 사용해야만 하는 이유가 있는 걸까요?
```suggestion
@RequiredArgsConstructor
public class QrService {

    private final RedisTemplate<String, byte[]> redisTemplate;
    private final MeetingRepository meetingRepository;
    private final MemberRepository memberRepository;
    private final MemberMeetingRepository memberMeetingRepository;
    private final MemberService memberService;
    private final MeetingService meetingService;
```

Reply:
아래 수정된 버전으로 바꿀게요